components/ide/jetbrains/launcher/log_analyzer.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package main
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"os/exec"
+	"regexp"
+
+	"github.com/gitpod-io/gitpod/common-go/log"
+)
+
+type LogAnalyzer interface {
+	Analyze(cmd *exec.Cmd, logFileName string) error
+	Stop()
+}
+
+const (
+	logDir                     = "/var/log/gitpod"
+	inCompatiblePatternLogFile = "jb-plugin-incompatible.log"
+)
+
+var (
+	inCompatiblePattern = regexp.MustCompile(`Plugin 'Gitpod Remote' .* is not compatible`)
+)
+
+// LauncherLogAnalyzer pipes JetBrains remote-dev-server.sh 's stdout and
+// stderr into os.Stdout and os.Stderr, and does diagnostic on the output
+type LauncherLogAnalyzer struct {
+	reader                    *io.PipeReader
+	writer                    *io.PipeWriter
+	inCompatibleBackendPlugin bool
+}
+
+var _ LogAnalyzer = &LauncherLogAnalyzer{}
+
+func (l *LauncherLogAnalyzer) Analyze(cmd *exec.Cmd, logFileName string) error {
+	reader, writer := io.Pipe()
+	l.reader = reader
+	l.writer = writer
+	stdout := io.MultiWriter(os.Stdout, writer)
+	stderr := io.MultiWriter(os.Stderr, writer)
+	cmd.Stdout = stdout
+	cmd.Stderr = stderr
+	go l.doAnalyze()
+	return nil
+}
+
+func (l *LauncherLogAnalyzer) Stop() {
+	l.writer.Close()
+}
+
+func (l *LauncherLogAnalyzer) doAnalyze() {
+	scanner := bufio.NewScanner(l.reader)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if !l.inCompatibleBackendPlugin && inCompatiblePattern.Match([]byte(line)) {
```

Comment:
can we detect somehow that our plugin is loaded as well? we had in the past issues that backend was started but without our plugin

Reply:
@akosyakov  I have another WIP (<3) https://github.com/gitpod-io/gitpod/pull/19642, it can find if our task terminals are attached via UI. With that test, we should be able to confirm if our plugin is loaded.

I'll check if there's anything logged by jb that our plugin is loaded

Reply:
I hoped we could parse some string where backend logs all loaded plugins and then parse whether our plugin is part of it?

Reply:
I want to have it as a metric at runtime, not only during integration tests.